File: 04_ruby/d01/Tristan_Marshall/reps.rb
################################ ROUND 1

def lengths(arrayOfStrings)
  arrayOfStrings.map { |s| s.length }
end

################################ ROUND 2

def transmorgifier(a, b, c)
  (a * b) ** c
end

################################ ROUND 3

def toonify(accent, sentence)
  if accent == 'daffy'
    return sentence.gsub 's', 'th'
  elsif accent == 'elmer'
    return sentence.gsub 'r', 'w'
  else
    return sentence
  end
end

################################ ROUND 4

def word_reverse(single)
  single.split(' ').reverse.join(' ')
end

################################ ROUND 5

def letterReverse(single)
  single.split(' ').map {}

################################ ROUND 6

File: 01_front_end_fundamentals/d04/Matt_Heron/reps.js

// Round 1
function lengths(stringList) {
    return stringList.map(function(s) {
       return s.length;
    });
}

//Round 2
function transmogrifier(a, b, c) {
  return Math.pow((a * b), c);
}

//Round 3
function toonify(accent, sentence) {
  if (accent === 'daffy') {
    return sentence.replace(/s/g, 'th');
  } else if (accent === 'elmer') {
    return sentence.replace(/r/g, 'w');
  } else {
    return sentence;
  }
}

//Round 4
function wordReverse(s) {
  var outString = "";
  var reversed = s.split(" ").reverse()
  for (var word=0; word < reversed.length; word++) {
    outString += reversed[word];
    if (word !== reversed.length) {
      outString += " ";
    }
  }
  return outString;
}

//Round 5
function letterReverse(s) {
  var outString = "";
  var stringArray = s.split(" ");
  for (var word=0; word < stringArray.length; word++) {
    var reversedStringArray = stringArray[word].split("").reverse();
    for (var character in reversedStringArray) {
      outString += reversedStringArray[character];
    }
    if (word !== stringArray.length) {
      outString += " ";
    }
  }
  return outString;
}

//Round 6
function longest(arrayOfStrings) {
  var longest = "";
  for (var item in arrayOfStrings) {
    if (arrayOfStrings[item].length > longest.length) {
      longest = arrayOfStrings[item];
    }
  }
  return longest;
}

//Final Round
function repMaster(input, inputFunction) {
  return inputFunction(input) + " proves that I am the rep MASTER!";
}
